API/RecommendationService/__init__.py

```diff
@@ -96,34 +98,42 @@ async def get_recommendation_items(command_list, recommend_type, error_info, cor
     knowledge_base_items_future = loop.run_in_executor(None, get_recommend_from_knowledge_base, command_list, recommend_type, error_info)
 
     # Get the recommendation of offline caculation from offline data
-    async def _get_offline_recommendation(command_list, recommend_type, error_info, command_top_num):
+    async def _get_offline_recommendation(command_list, recommend_type, command_top_num):
         offline_items = []
-        if need_offline_recommendation(recommend_type, error_info=None):
-            offline_items = await get_recommend_from_offline_data(command_list, recommend_type, error_info=None, top_num=command_top_num)
+        if need_offline_recommendation(recommend_type):
+            offline_items = await get_recommend_from_offline_data(command_list, recommend_type, top_num=command_top_num)
         return offline_items
-    calculation_items_future = _get_offline_recommendation(command_list, recommend_type, error_info, command_top_num)
+    calculation_items_future = _get_offline_recommendation(get_success_commands(command_list), recommend_type, command_top_num)
```

Comment:
It is recommended to define a variable for `get_success_commands(command_list)` to reuse